src/kubelet/client/client_mock.go

```diff
@@ -0,0 +1,57 @@
+package client
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/newrelic/infra-integrations-sdk/log"
+
+	"github.com/newrelic/nri-kubernetes/v2/src/client"
+	"github.com/newrelic/nri-kubernetes/v2/src/prometheus"
+)
+
+// Client implements a client for Kubelet, capable of retrieving prometheus metrics from a given endpoint.
+type Mock struct {
```

Comment:
How does this mock compare to using the real client implementation, but with the `testutil` fake server as a backend?

Reply:
I removed that mock, but I kept the connector mock for now, mainly because when testing the client setting up all objs needed to perform discovery is a bit annoying (/healtz page), node with IP of the server, etc